Use existing docker machine if available

zerodivisi0n opened the pull request:
Does not create new VirtualBox docker machine with `default` name if another machine already exists. Uses the first available machine from the `docker-machine ls`. Allows to use machines created with different drivers therefore partially solves the issue #231.

FrenchBen commented on src/utils/SetupUtil.js:
```diff
@@ -115,31 +115,50 @@ export default {
+          }
+        }
+
+        if (!exists) {
```
Why the 2 `if` cases - these should be combined.

zerodivisi0n replied:
Wanted to keep checking logic more straightforward. I combined it now.

FrenchBen commented:
Thank you for the PR - Could you make sure to sign your commit, as seen in the contribution guidelines?
https://github.com/docker/docker/blob/master/CONTRIBUTING.md#sign-your-work

Although I like the idea of selecting the first Machine available, this would break some of the existing functionality if the first machine was a remote one. Volume mounting being the first one to come to mind. 
I would filter your 'ls' to look for machines that have a specific driver known to work and local.

zerodivisi0n commented:
Added filter to work only with local drivers. List of drivers I get from [Supported Drivers](https://docs.docker.com/machine/drivers/) and [Available driver plugins](https://github.com/docker/machine/blob/master/docs/AVAILABLE_DRIVER_PLUGINS.md). I don't know how to fully check their workability, so I added only local and active.

Also signed my commits.

FrenchBen commented:
Finally circling back to this, sorry for the delay.

2 things came to mind:
1. Why save the Machine name? I may create a default after initial launch and can't easily reset it
2. No visual indication of machines - The user has no visual queue that the 'default' machine isn't being used and that some other machine is being loaded. 

Allowing the user to select his/her machine would make more sense IMHO. Although pretty old now, it's what we discussed in the early https://github.com/docker/kitematic/wiki/Machine-Management docs
